crates/floresta-wire/src/p2p_wire/running_node.rs

```diff
@@ -358,8 +359,9 @@ where
 
         info!("starting running node...");
         loop {
-            while let Ok(notification) =
-                timeout(Duration::from_millis(100), self.node_rx.recv()).await
+            while let Some(notification) = timeout(Duration::from_millis(100), self.node_rx.recv())
+                .await
+                .unwrap()
```

Comment:
This unwrap is causing my node to crash. Timeout is returned as `Err` here. Change the patter matching to `Ok(Some(notification))`.